dataprofiler/labelers/character_level_cnn_model.py

```diff
@@ -304,6 +305,30 @@ def __init__(self, label_mapping=None, parameters=None):
         :return: None
         """
 
+        # https://github.com/tensorflow/tensorflow/issues/34697
+        def make_keras_picklable():
+            def __getstate__(self):
+                model_str = ""
+                with tempfile.NamedTemporaryFile(suffix='.hdf5', delete=True) as fd:
+                    tf.keras.models.save_model(self, fd.name, overwrite=True)
+                    model_str = fd.read()
+                d = {'model_str': model_str}
+                return d
+                
+            def __setstate__(self, state):
+                with tempfile.NamedTemporaryFile(suffix='.hdf5', delete=True) as fd:
+                    fd.write(state['model_str'])
+                    fd.flush()
+                model = tf.keras.models.load_model(fd.name)
+                self.__dict__ = model.__dict__
+
+            cls = tf.keras.models.Model
+            cls.__getstate__ = __getstate__
+            cls.__setstate__ = __setstate__
+
+        # Run the function
+        make_keras_picklable()    
```

Comment:
Some questions purely because I'm curious:
Okay so if I understand this correctly, you are overwriting the get and set state for a keras model so that it is in the form of a dictionary so that you can pickle it, but what I don't understand is how does the tempfile work? like how is the tempfile going to be replaced by the actual model? Is that just how the tempfile library works? it is designed to get replaced with whatever is in the keras model?
Secondly, what about models that dont have the .hdf5 format?
Thirdly, are models that have already been created, still going to be pickleable?
Fourthly, is there ever a time when you wouldnt want the model to be pickleable?